Fix search input styling in Safari

kinoushe opened the pull request:
Closes #2885 

Changes proposed in this pull request:

- Disable browser-native search input styling in Safari to be consistent with other web browsers

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested
- [ ] unit tests written (optional if too hard)
- [ ] documentation updated

Alkarex commented:
Thanks @kinoushe 👍 
Could you please put those rules in
* https://github.com/FreshRSS/FreshRSS/blob/master/p/themes/base-theme/template.css
* https://github.com/FreshRSS/FreshRSS/blob/master/p/themes/base-theme/template.rtl.css

Those files are used by other themes, while `base.css` is just an example of basic theme to actually used.

Alkarex commented:
For reference https://developer.mozilla.org/docs/Web/CSS/appearance
@Frenzie Could you please check what we can do with Travis regarding the CSS linting error here?

Frenzie commented:
I think I'd prefer not to make a global exception but you can just stick `/* csslint allow: prefix-whatever-its-called */` behind the line (or behind the `{` for the whole block).

kinoushe commented:
Sorry @Alkarex I couldn't find those selectors in the `template.css` files, so have now created them and reverted the `base.css` changes.

@Frenzie I've added the `/* csslint allow: -webkit-appearance */` comment, please let me know if this is incorrect.

Thanks for your patience, I'm very new to this!

Alkarex commented on p/themes/base-theme/template.css:
```diff
@@ -572,6 +572,10 @@ a.btn {
 }
 
+.header > .item.search input {
+	-webkit-appearance: none; /* csslint allow: -webkit-appearance */
```
```suggestion
.header > .item.search input {
	/* stylelint-disable property-no-vendor-prefix */
	-webkit-appearance: none;
	/* stylelint-enable property-no-vendor-prefix */
}
```

There might be a better syntax, but this one seems to work

Alkarex commented on p/themes/base-theme/template.css:
```diff
@@ -572,6 +572,10 @@ a.btn {
 }
 
+.header > .item.search input {
+	-webkit-appearance: none; /* csslint allow: -webkit-appearance */
```
```suggestion
	/* stylelint-disable property-no-vendor-prefix */
	-webkit-appearance: none;
	/* stylelint-enable property-no-vendor-prefix */
```

There might be a better syntax, but this one seems to work

Frenzie replied:
I meant `-webkit-appearance: none; /* csslint allow: property-no-vendor-prefix */` btw.

Not that it matters or should be changed or anything, I just think it looks slightly neater. ;-)

Alkarex replied:
Please send a fix if you have a better syntax @Frenzie , so that it can also be used as a reference for future needs

Alkarex commented on p/themes/base-theme/template.rtl.css:
```diff
@@ -572,6 +572,10 @@ a.btn {
 }
 
+.header > .item.search input {
+	-webkit-appearance: none; /* csslint allow: -webkit-appearance */
```
```suggestion
	/* stylelint-disable property-no-vendor-prefix */
	-webkit-appearance: none;
	/* stylelint-enable property-no-vendor-prefix */
```

Alkarex commented on p/themes/base-theme/template.css:
```diff
@@ -572,6 +572,12 @@ a.btn {
 	width: 100px;
 }
 
+.header > .item.search input {
```
Maybe we should go for a faster & better selector. Could you try that one:

```suggestion
search[type="search"] {
```

kinoushe replied:
Yes, works as expected in Safari and would cover any other search inputs.

Alkarex commented on p/themes/base-theme/template.rtl.css:
```diff
@@ -572,6 +572,12 @@ a.btn {
 	width: 100px;
 }
 
+.header > .item.search input {
```
```suggestion
search[type="search"] {
```

Alkarex commented:
Is it only Safari that is affected, or other Webkit-based browsers? And which platforms?

kinoushe commented:
Safari is the only Webkit browser requiring the style fix that I can test. I've checked the latest versions of Chrome and Opera and they're unaffected by the fix. I can only test on a Mac.

kinoushe commented:
It also corrects the display in Safari on iPadOS/iOS where the search input appears as a pill shape otherwise:

![IMG_5177](https://user-images.githubusercontent.com/25278192/78990295-76a4ea00-7b79-11ea-80f8-60637af995d9.png)
![IMG_5178](https://user-images.githubusercontent.com/25278192/78990301-799fda80-7b79-11ea-8da8-c1106942f0eb.png)

Alkarex commented:
Ok, thanks, and please put a line for you in https://github.com/FreshRSS/FreshRSS/blob/master/CREDITS.md in another PR 👍
